lib/Transforms/MemoryBanking.cpp

```diff
@@ -201,8 +201,27 @@ SmallVector<Value, 4> handleGetGlobalOp(memref::GetGlobalOp getGlobalOp,
 }
 
 SmallVector<Value, 4> createBanks(Value originalMem, uint64_t bankingFactor,
-                                  unsigned bankingDimension) {
+                                  std::optional<unsigned> bankingDimensionOpt) {
   MemRefType originalMemRefType = cast<MemRefType>(originalMem.getType());
+  unsigned rank = originalMemRefType.getRank();
+  ArrayRef<int64_t> shape = originalMemRefType.getShape();
+
+  // Get the banking dimension if specified; use the innermost dimension that
+  // has size greater than one if not.
+  unsigned bankingDimension;
```

Comment:
Its not obvious to me this will always be assigned a value. Can you provide it some default value, or make this optional and assert it has a value?

Reply:
sounds good, done!